tests/Backend/NetworkPolicies/NetworkPoliciesTestCase.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Test\Backend\NetworkPolicies;
+
+use Doctrine\DBAL\Exception\DriverException;
+use Keboola\StorageDriver\Shared\Utils\StringUtils;
+use Keboola\Test\StorageApiTestCase;
+use Keboola\Test\Utils\ConnectionUtils;
+use Keboola\Test\Utils\IpUtils;
+use Ramsey\Uuid\Nonstandard\Uuid;
+use Throwable;
+
+class NetworkPoliciesTestCase extends StorageApiTestCase
+{
+    use ConnectionUtils;
+    use IpUtils;
+
+    private function getDBPrefix(): string
+    {
+        $dbPrefix = getenv('SNOWFLAKE_DB_PREFIX');
+        assert($dbPrefix !== false, 'SNOWFLAKE_DB_PREFIX env var is not set');
+        return $dbPrefix;
+    }
+
+    private function useRoleAccountAdmin(): void
```

Comment:
nemam to projity jeste vsechno, ale premyslim, jestli by metoda, co vola `useRoleAccountAdmin` nemela volat i neco jako `userPreviousRole`. Jde o to, ze ted nekdo na zacatku zavola `useRoleAccountAdmin` a pak cele spojeni uz jede s ACCOUNTADMIN a najednou muzes testovat neco se muze chovat jinak s accountadmin a jinou roli

Reply:
já to bral tak, že tohle je jen servisní account (NETWORK_POLICIES_KBC_STORAGE) který se reálně nepoužívá na nic jiného než tady v testu na tu manipulaci s NETWORK RULES a zjišťování stavu NETWORK POLICY

takže nevadí, že bude mít pořád ACCOUNTADMIN

ale pro větší pocit sucha a bezpečí můžu udělat něco jako

```
$this->callWithAccountAdmin(callable $callback);
```

Které na začátku zjistí aktuální roli, nastaví accountadmin a pak vrátí tu předchozí roli

Reply:
no ta moje poznanka vychazi z toho, ze by mohly existovat situace, kdy mas neco nastavit jako accountadmin a pak udelat operaci co accountadmina nepotrebuje. ale tady v tom testu to zatim tak nevypada, takze asi cajk. mozna jen prosim do komentare k teto metode neco dopis, ze takove nebezpeci existuje

Reply:
ten callback je asi nejklidnejsi reseni, ale asi zbytecne povyku v teto chvili

Reply:
já tvou poznámku úplně chápu a píšu, že jsem to vzal v potaz a máš pravdu :-)

jen jen pro tenhle případ a test je to fuk, že celé spojení bude pod accountadmin (stejně to tak chceme) a pokud někdo tenhle test zneužije jinde nějak blbě, tak by jsme to teprve měli řešit

Reply:
alright